src/site/content/en/blog/monitor-and-analyze/index.md

```diff
@@ -0,0 +1,252 @@
+
+<figure>
+{% Img src="image/T4FyVKpzu4WKF1kBNvXepbi08t52/WpIg5tBzxAXzEZtPMS1Y.png", alt="A screenshot of the bundlesize output. Some build
+results are marked with 'Fail'", width="800", height="590" %}
+</figure>
+
+### Further reading
+
+- Alex Russell [about the real-world loading time we should
+target](https://infrequently.org/2017/10/can-you-afford-it-real-world-web-performance-budgets/)
+
+## Analyze why the bundle is so large
+
+You might want to dig deeper into the bundle to see what modules take space in it. Meet
+[webpack-bundle-analyzer](https://github.com/webpack-contrib/webpack-bundle-analyzer):
+
+<figure>
+  {% Video src="video/T4FyVKpzu4WKF1kBNvXepbi08t52/fXNWhHRJMRe0ELwJPsd4.mp4", autoplay="true", loop="true", muted="true" %}
+  <figcaption>(Screen recording from <a
+href="https://github.com/webpack-contrib/webpack-bundle-analyzer">github.com/webpack-contrib/webpack-bundle-analyzer</a>)
+</figcaption>
+</figure>
+
+webpack-bundle-analyzer scans the bundle and builds a visualization of what’s inside it. Use this
+visualization to find large or unnecessary dependencies.
+
+To use the analyzer, install the `webpack-bundle-analyzer` package:
+
+```shell
+npm install webpack-bundle-analyzer --save-dev
+```
+
+add a plugin to the webpack config:
+
+```js
+// webpack.config.js
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+
+module.exports = {
+    plugins: [
+    new BundleAnalyzerPlugin(),
+    ],
+};
+```
+
+and run the production build. The plugin will open the stats page in a browser.
+
+By default, the stats page shows the size of parsed files (i.e., of files as they appear in
+the bundle). You’ll likely want to compare gzip sizes since that’s closer to what real users
+experience; use the sidebar on the left to switch the sizes.
+
+{% Aside %}
+If you use the
+[ModuleConcatenationPlugin](https://webpack.js.org/plugins/module-concatenation-plugin/), it might
+merge a part of modules in the webpack-bundle-analyzer output, making the report less detailed. If
+you use this plugin, disable it during the analysis.
+{% endAside %}
+
+Here’s what to look for in the report:
+
+- **Large dependencies.** Why are they so large? Are there smaller alternatives (e.g., Preact
+instead of React)? Do you use all the code it includes (e.g., Moment.js includes a lot of locales
+[that are often not used and could be dropped](https://github.com/GoogleChromeLabs/webpack-libs-optimizations#moment))?
+
+- **Duplicated dependencies.** Do you see the same library repeating in multiple files? (Use, e.g.,
+the `optimization.splitChunks.chunks` option – in webpack 4 – or the `CommonsChunkPlugin` –
+in webpack 3 – to move it into a common file.) Or does the bundle have multiple versions
+of the same library?
+
+- **Similar dependencies.** Are there similar libraries that do approximately the same job? (E.g.
+`moment` and `date-fns`, or `lodash` and `lodash-es`.) Try sticking with a single tool.
```

Comment:
```suggestion
- **Large dependencies.** Why are they so large? Are there smaller alternatives (e.g., Preact
instead of React)? Do you use all the code it includes (e.g., Moment.js includes a lot of locales
[that are often not used and could be dropped](https://github.com/GoogleChromeLabs/webpack-libs-optimizations#moment))?
- **Duplicated dependencies.** Do you see the same library repeating in multiple files? (Use, e.g.,
the `optimization.splitChunks.chunks` option – in webpack 4 – or the `CommonsChunkPlugin` –
in webpack 3 – to move it into a common file.) Or does the bundle have multiple versions
of the same library?
- **Similar dependencies.** Are there similar libraries that do approximately the same job? (E.g.
`moment` and `date-fns`, or `lodash` and `lodash-es`.) Try sticking with a single tool.
```